packages/x-date-pickers-pro/src/DesktopDateRangePicker/tests/DesktopDateRangePicker.test.tsx

```diff
@@ -333,56 +364,58 @@ describe('<DesktopDateRangePicker />', () => {
         />,
       );
 
-      openPicker({ type: 'date-range', initialFocus: 'start', fieldType: 'single-input' });
+      await openPickerAsync(user, {
+        type: 'date-range',
+        initialFocus: 'start',
+        fieldType: 'single-input',
+      });
 
       // Change the start date (already tested)
-      fireEvent.click(getPickerDay('3'));
+      await user.click(getPickerDay('3'));
 
       // Dismiss the picker
-      // eslint-disable-next-line material-ui/disallow-active-element-as-key-event-target -- don't care
-      fireEvent.keyDown(document.activeElement!, { key: 'Escape' });
+      await user.keyboard('[Escape]');
       expect(onChange.callCount).to.equal(1); // Start date change
       expect(onAccept.callCount).to.equal(1);
       expect(onAccept.lastCall.args[0][0]).toEqualDateTime(new Date(2018, 0, 3));
       expect(onAccept.lastCall.args[0][1]).toEqualDateTime(defaultValue[1]);
       expect(onClose.callCount).to.equal(1);
     });
 
-    it('should call onClose when clicking outside of the picker without prior change (multi input field)', () => {
+    it('should call onClose when clicking outside of the picker without prior change (multi input field)', async () => {
       const onChange = spy();
       const onAccept = spy();
       const onClose = spy();
 
-      render(
+      const { user } = render(
         <div>
+          <input id="test-id" />
           <DesktopDateRangePicker
             onChange={onChange}
             onAccept={onAccept}
             onClose={onClose}
             slots={{ field: MultiInputDateRangeField }}
           />
-          <input id="test-id" />
         </div>,
       );
 
-      openPicker({ type: 'date-range', initialFocus: 'start', fieldType: 'multi-input' });
+      await openPickerAsync(user, {
+        type: 'date-range',
+        initialFocus: 'start',
+        fieldType: 'multi-input',
+      });
 
       // Dismiss the picker
       const input = document.getElementById('test-id')!;
 
-      fireEvent.mouseDown(input);
-      act(() => {
-        input.focus();
-      });
-      fireEvent.mouseUp(input);
-      clock.runToLast();
+      await user.click(input);
 
       expect(onChange.callCount).to.equal(0);
       expect(onAccept.callCount).to.equal(0);
-      expect(onClose.callCount).to.equal(1);
+      expect(onClose.callCount).to.equal(2);
```

Comment:
I have no idea why it fires 2 close on test. 

If we change the `user.click` above to a `user.keyboard('[Escape]')` it works correctly. If we add a `user.click(getPickerDay('3'))` like the test below it works correctly also.

😢

Reply:
Thanks for the suggestion, using `[MouseLeft>]` seems to work, though it is quite odd